subt/odoimuloc.py

```diff
@@ -0,0 +1,80 @@
+import collections
+import logging
+import math
+import osgar.node
+
+import osgar.lib.quaternion as quaternion
+
+g_logger = logging.getLogger(__name__)
+
+Pose2d = collections.namedtuple("Pose2d", ("x", "y", "heading"))
+
+class Localization(osgar.node.Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        # outputs
+        bus.register('pose3d')
+        # inputs: origin, orientation, odom
+        self.xyz = None
+        self.orientation = None
+        self.last_odom = None
+        self.origin_xyz = None
+        self.origin_orientation = None
+        self.origin_error = False
+
+    def on_origin(self, data):
+        if len(data) == 8:
+            self.xyz = data[1:4]
+            self.origin_xyz = data[1:4]
+            self.origin_orientation = data[4:]
+        else:
+            self.origin_error = True
+
+    def on_orientation(self, dt, orientation):
```

Comment:
we should unify these `on_<something>` "template" so that we can reused the caller (like in `moon.MoonNode`). `on_origin` has one parameter, `on_orientation` two and `on_odom` even more ... is it intended?

Reply:
updated to use what we used for slots - `dt` + `data` returned from `bus.listen()`

Reply:
I will add here note before I forgot it - we agreed on call to use unified "data" only, i.e. to remove timestamp. Ideally in the future the data would also contain internally when exactly they were acquired. As a side-effect the `on_something` callback will be the same as in Moon project now.